File: Crio/crio_programming_interview_problems-master/MeetingRooms/MeetingRooms.cpp
#include<bits/stdc++.h>
using namespace std;

int meetingRooms(int n, vector<vector<int>>& time){
    // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    vector<pair<int,int>>meeting;
    for (auto &&m : time)
    {
        meeting.push_back({m[0],m[1]});
    }
    sort(meeting.begin(), meeting.end());
    priority_queue<int,vector<int>,greater<int>>q;
    int ans = 0;
    for (int i = 0; i < n; i++)
    {
        int start = meeting[i].first;
        int end = meeting[i].second;
        while(q.size()>0 && q.top()<=start){
            q.pop();
        }
        q.push(end);
        ans = max(ans, (int)(q.size()));
    }
    return ans;
    // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
}
int main(){
    int n;
    cin>>n;
    vector<vector<int>>time(n);
    for (int i = 0; i < n; i++)
    {
        int s,e;
        cin>>s>>e;
        time[i].push_back(s);
        time[i].push_back(e);
    }
    int ans = meetingRooms(n,time);
    cout<<ans;
}

File: Crio/crio_programming_interview_problems-master/MeetingRooms/MeetingRooms.java
import java.util.*;

public class MeetingRooms {

    public int findNumRooms(int[][] intervals) {
            // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
	    if (intervals.length == 0) {
	      return 0;
	    }
	    // Min Heap to hold the ending time
	    PriorityQueue<Integer> minHeap = new PriorityQueue<Integer>(
		                                 intervals.length,
		                                 new Comparator<Integer>() {
		                                    public int compare(Integer a, Integer b) {
			                            return a - b;
		                                 }});

	    // Sort intervals by starting time
	    Arrays.sort(intervals, new Comparator<int[]>() {
		                      public int compare(final int[] a, final int[] b) {
		                          return a[0] - b[0];
		                      }});

	    // Add first meeting to minHeap to start with
	    minHeap.add(intervals[0][1]);

	    for (int i = 1; i < intervals.length; i++) {
	      if (intervals[i][0] >= minHeap.peek()) {  
		minHeap.poll();
	      }
	      minHeap.add(intervals[i][1]);
	    }

	    return minHeap.size();  // This is the min rooms required
            // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
    }

    public static void main(String[] args)
    {
        Scanner scanner = new Scanner(System.in);

        int n = scanner.nextInt();

        int[][] nums = new int[n][2];

        for(int i = 0 ; i < n ;i++) {
            nums[i][0] = scanner.nextInt();
            nums[i][1] = scanner.nextInt();
        }

        int result = new MeetingRooms().findNumRooms(nums);

        System.out.printf("%d", result);
    }

}

